lib/facter/splunkenterprise.rb

```diff
@@ -1,17 +1,18 @@
-Facter.add(:splunk_version) do
+Facter.add(:splunkenterprise) do
   setcode do
-    value = nil
+    splunk_hash = {}
```

Comment:
I've never encountered another fact that returns an empty hash like this. The behavior I'd expect is that if there is no data related to splunk then the fact should be undefined within a manifest. If the behavior isn't followed then this fact will exist on all nodes within an infrastructure due to the way pluginsync works.

Reply:
Agreed, I'll make some changes